src/redux/dragon/dragonSlice.js

```diff
@@ -0,0 +1,43 @@
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
+import axios from "axios";
+
+const url = "https://api.spacexdata.com/v4/dragons";
+
+export const getDragon = createAsyncThunk(
+  'books/getDragons',
+  async () => {
+    const res = await axios.get(
+      url,
+    )
+    const dragons = Object.keys(res.data).map((key) => ({
+      mission_id: key,
+      ...res.data[key]
+    }))
+    return dragons;
+  }
+)
+
+const initialState = {
+  dragonStore: [],
+  status: 'idle',
+  error: '',
+}
+
+const dragonSlice = createSlice({
```

Comment:
- [ ] kindy Once the data are fetched, dispatch an action to store the selected data in Redux store:

-id
-name
-type
-flickr_images

- [ ] Make sure you only dispatch those actions once and do not add data to store on every re-render (i.e. when changing views / using navigation).